appengine/genetics/js/visualization.js

```diff
@@ -324,187 +535,218 @@ Genetics.Visualization.processCageEvents_ = function() {
   var getMouseName = Genetics.Visualization.getMouseName;
   while (Genetics.Cage.EVENTS.length) {
     var event = Genetics.Cage.EVENTS.shift();
+
+    var mouse = (event['ID'] != null) ?
+        Genetics.Visualization.mice_[event['ID']] : null;
```

Comment:
I see.  Since we are making a distinction between null and 0, it would be good to use !== here.